File: app/src/main/java/com/zillennium/utswap/screens/setting/MyWallet/MyWalletGroupAdapter.java
package com.zillennium.utswap.screens.setting.MyWallet;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.zillennium.utswap.R;

import java.util.ArrayList;

public class MyWalletGroupAdapter extends RecyclerView.Adapter<MyWalletGroupAdapter.ViewHolder> {
    private ArrayList<MyWalletGroup> arrayList = new ArrayList<>();

    public <onclickBalanceHistory> MyWalletGroupAdapter(ArrayList<MyWalletGroup> arrayList) {
        this.arrayList = arrayList;
    }

    @NonNull
    @Override
    public MyWalletGroupAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_list_wallet_balance_group,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyWalletGroupAdapter.ViewHolder holder, int position) {
        MyWalletGroup myWalletGroup = arrayList.get(position);

        holder.txtDayGroup.setText(myWalletGroup.getSelectDayGroup());

        MyWalletItemAdapter myWalletItemAdapter = new MyWalletItemAdapter(myWalletGroup.getSelectItemGroup());
        holder.rcItemGroupDay.setAdapter(myWalletItemAdapter);

    }

    @Override
    public int getItemCount() {
        return arrayList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        private final TextView txtDayGroup;
        private final RecyclerView rcItemGroupDay;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            txtDayGroup = itemView.findViewById(R.id.txt_day_group);
            rcItemGroupDay = itemView.findViewById(R.id.rc_item_group_day);


        }
    }
}

File: app/src/main/java/com/zillennium/utswap/screens/wallet/transactionScreen/adapter/MyWalletGroupAdapter.kt
package com.zillennium.utswap.screens.wallet.transactionScreen.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.zillennium.utswap.R
import com.zillennium.utswap.screens.wallet.transactionScreen.model.MyWalletGroup

class MyWalletGroupAdapter(arrayList: ArrayList<MyWalletGroup>) :
    RecyclerView.Adapter<MyWalletGroupAdapter.ViewHolder>() {
    private var arrayList: ArrayList<MyWalletGroup> = ArrayList<MyWalletGroup>()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view: View = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_list_wallet_balance_group, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val myWalletGroup: MyWalletGroup = arrayList[position]
        holder.txtDayGroup.text = myWalletGroup.selectDayGroup
        val myWalletItemAdapter = MyWalletItemAdapter(myWalletGroup.selectItemGroup)
        holder.rcItemGroupDay.adapter = myWalletItemAdapter
    }

    override fun getItemCount(): Int {
        return arrayList.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val txtDayGroup: TextView = itemView.findViewById(R.id.txt_day_group)
        val rcItemGroupDay: RecyclerView = itemView.findViewById(R.id.rc_item_group_day)

    }

    init {
        this.arrayList = arrayList
    }
}